snuba/datasets/factory.py

```diff
@@ -1,10 +1,9 @@
 from snuba import settings
-from snuba.datasets.events import EventsDataSet
 
 DATASETS_IMPL = {}
 
-DATASETS_MAPPING = {
-    'events': EventsDataSet,
+DATASET_NAMES = {
```

Comment:
This change is only meant to make circular imports less of a problem. 
Before this change a lot of places imported this module and this one has to import all datasets. This made it hard to avoid circular imports. Now we only import datasets when we need to instantiate them.

Reply:
In general the dataset class hierarchy should not depend on the factory that creates datasets. This design is already going in that direction. The circular dependency is that the processor depends on util, util depends on the query json schema (not the table schema) and that depends on the dataset names enum (just the names) which is here. I think the right way is to split the dataset enum from the the dataset factory. It just won't be obvious to whoever reads the code why we did that